public/js/EventsPage.js

```diff
@@ -0,0 +1,78 @@
+//Adds event listener to submit button that gets data from form and validates it.
+document.getElementById('submit-event').addEventListener('click', e => {
+    e.preventDefault()
+    let data = getCompletedFormData()
+    let validate = validateForm()
+    if(validate) {
+        //This is where the object of the form data is sent if valid data.
+        //It's console logged now for demonstration.
+       console.log(data)
+    }
+})
+
+function validateForm() {
+    let success = true
+    let message = ''
+    let inputs = document.querySelectorAll('.create-events')
+    inputs.forEach(function (element) {
+        //Checks fields with attribute data-required=true has data
+        let required = element.getAttribute('data-required')
+        if (required && element.value.length < 1) {
+            message += element.previousElementSibling.innerHTML + ' is a required field! <br>'
+            success = false
+        }
```

Comment:
In the interests of DRY and code-not-falling-over, I would have a separate validation function for each field. This way if there's a problem with the validation across mutiple fields all of those errors can be shown at once / not blocked waiting for previous fields to validate.

Reply:
being done on seperate branch